tests/client_test.py

```diff
@@ -610,3 +630,317 @@ def test_intercept_unary_unary_failed_request(self):
+    """Tests for the google.ads.googleads.client.ExceptionInterceptor class."""
+
+    _MOCK_FAILURE_VALUE = b"\n \n\x02\x08\x10\x12\x1aInvalid customer ID '123'."
+
+    def _create_test_interceptor(self):
+        return google.ads.google_ads.client.ExceptionInterceptor()
+
+    def test_init_(self):
+        interceptor = self._create_test_interceptor()
+        self.assertEqual(interceptor._RETRY_STATUS_CODES,
+            (grpc.StatusCode.INTERNAL, grpc.StatusCode.RESOURCE_EXHAUSTED))
+
+    def test_get_request_id(self):
+        mock_metadata = (('request-id', '123456'),)
+        interceptor = self._create_test_interceptor()
+        result = interceptor._get_request_id(mock_metadata)
+        self.assertEqual(result, '123456')
+
+    def test_get_request_id_no_id(self):
+        mock_metadata = (('another-key', 'another-val'),)
+        interceptor = self._create_test_interceptor()
+        result = interceptor._get_request_id(mock_metadata)
+        self.assertEqual(result, None)
+
+    def test_get_google_ads_failure(self):
+        interceptor = self._create_test_interceptor()
+        mock_metadata = ((interceptor._FAILURE_KEY, self._MOCK_FAILURE_VALUE),)
+        result = interceptor._get_google_ads_failure(mock_metadata)
+        self.assertIsInstance(result, error_protos.GoogleAdsFailure)
+
+    def test_get_google_ads_failure_decode_error(self):
+        interceptor = self._create_test_interceptor()
+        mock_failure_value = self._MOCK_FAILURE_VALUE + b'1234'
+        mock_metadata = ((interceptor._FAILURE_KEY, mock_failure_value),)
+        result = interceptor._get_google_ads_failure(mock_metadata)
+        self.assertEqual(result, None)
+
+    def test_get_google_ads_failure_no_failure_key(self):
+        mock_metadata = (('another-key', 'another-val'),)
+        interceptor = self._create_test_interceptor()
+        result = interceptor._get_google_ads_failure(mock_metadata)
+        self.assertEqual(result, None)
+
+    def test_get_google_ads_failure_with_None(self):
+        interceptor = self._create_test_interceptor()
+        result = interceptor._get_google_ads_failure(None)
+        self.assertEqual(result, None)
+
+    def test_handle_grpc_exception(self):
+        """If exception is not retryable and is a GoogleAdsFailure convert to
+           GoogleAdsException and raise.
+        """
+        mock_error_message = self._MOCK_FAILURE_VALUE
+
+        class MockRpcError(grpc.RpcError):
+            def code(self):
+                return grpc.StatusCode.INVALID_ARGUMENT
+
+            def trailing_metadata(self):
+                return ((interceptor._FAILURE_KEY, mock_error_message),)
+
+        interceptor = self._create_test_interceptor()
+
+        self.assertRaises(
+            GoogleAdsException,
+            interceptor._handle_grpc_exception,
+            MockRpcError()
+        )
+
+    def test_handle_grpc_exception_retryable(self):
+        """If exception is retryable raise as-is.
+        """
```

Comment:
The triple-quotes should go on the previous line.

Reply:
Fixed.